octodns/manager.py

```diff
@@ -331,6 +352,25 @@ def sync(self, eligible_zones=[], eligible_sources=[], eligible_targets=[],
         # plan pairs.
         plans = [p for f in futures for p in f.result()]
 
+        # Populate aliases zones.
+        futures = []
+        for zone_name, zone_source in aliased_zones.items():
+            plan = [p for t, p in plans if p.desired.name == zone_source]
+            if not plan:
```

Comment:
I have not found how to generate an empty plan in order to test this condition.

Reply:
> Nope, at least not im my tests. I've created a record called `only-in-dot-net-target` in my alias zone at DNS Made Easy and it's not deleted.

This is actually the issue that's preventing there from being a run/plan on the aliased zone when there are no changes in the original. 

See https://gist.github.com/ross/a154334eb3a121a9240ba9c52f1c637c for the miniaml test case i did to set things up and test changes to the config. All that works as expected.

When I go into the zone in Route53 and manually add a record to aliased.net. Though it isn't detected as you saw. Reason for this is that orignal.net didn't have any changes so there's no plan for it. We come into this loop and get back no plans and come into this if and skip planning for aliased.net and don't see that it needs the extra record deleted (or any other changes that it might need.) 

So this if is why it's not even trying, but the problem is a little deeper as you point out since we won't have a copy of the desired state to pass in. Going to poke at this a bit.

Reply:
https://github.com/github/octodns/commit/661cbb2cd255fe4259e7b5130f3a0e5c8859c964 should have the fix for the `desired` issue. Basically it modifies `_populated_and_plan` to always returne the (desired) zone so that it can be stored and used for aliased zones if needed. 

You should be able to cherry pick that in with something like:

```console
$ git fetch origin
$ git cherry-pick 661cbb2cd255fe4259e7b5130f3a0e5c8859c964
```

I would have PR'd against yours so you could just merge it, but looks like that's not supported across forks.

Reply:
I've applied you patch and everything woks perfectly now. Thank you!